src/Token/TokenLabel.tsx

```diff
@@ -0,0 +1,178 @@
+        }
+      `
+    }
+  }}
+`
+
+const TokenTextContainer = styled('span')`
+  flex-grow: 1;
+  min-width: 0;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`
+
+const TokenLabel = forwardRef<HTMLElement, TokenLabelProps>((props, forwardedRef) => {
+  const {as, fillColor, handleRemove, id, isSelected, ref, text, variant, hideRemoveButton, ...rest} = props
+  const {colorScheme} = useTheme()
+  const {bgOpacity, borderOpacity, borderThreshold, lightnessThreshold} = colorModeConfigs[colorScheme || 'light']
+  let bgColor = fillColor
+  let borderColor = fillColor
+  let textColor = '#FFF'
+  const perceivedLightness = tinycolor(fillColor).getLuminance()
+  const isFillColorLight = perceivedLightness >= lightnessThreshold
+
+  if (colorScheme === 'dark') {
+    const lightenBy = (perceivedLightness - lightnessThreshold) * 100 * (isFillColorLight ? 1 : 0)
+
+    bgColor = isSelected
+      ? tinycolor(fillColor)
+          .setAlpha(bgOpacity * 1.2)
+          .toRgbString()
+      : tinycolor(fillColor).setAlpha(bgOpacity).toRgbString()
+    textColor = isSelected
+      ? tinycolor(fillColor)
+          .lighten(lightenBy + 10)
+          .toString()
+      : tinycolor(fillColor).lighten(lightenBy).toString()
+    borderColor = isSelected
+      ? tinycolor(fillColor)
+          .lighten(lightenBy + 5)
+          .toRgbString()
+      : tinycolor(fillColor).lighten(lightenBy).setAlpha(borderOpacity).toRgbString()
+  } else {
+    const isFillColorDark = perceivedLightness < 0.1
+    borderColor = perceivedLightness >= borderThreshold ? tinycolor(fillColor).darken(25).toString() : 'transparent'
+
+    if (isFillColorLight) {
+      textColor = '#000'
+    }
+
+    if (isSelected) {
+      bgColor = isFillColorDark
+        ? tinycolor(fillColor).lighten(10).toString()
+        : // TODO: darken more than 10 if the fillColor is really bright and doesn't darken well
+          //       Examples of colors that don't darken well:
+          //       - #a2eeef
+          //       - #ffd78e
+          //       - #a4f287
+          tinycolor(fillColor).darken(10).toString()
+    }
+  }
+
+  const hasMultipleActionTargets = isTokenInteractive(props) && Boolean(handleRemove) && !hideRemoveButton
+  const handleRemoveClick: MouseEventHandler = e => {
+    e.stopPropagation()
+    handleRemove && handleRemove()
+  }
+
+  return (
+    <StyledTokenLabel
+      // specific to labels
+      fillColor={fillColor}
+      bgColor={bgColor}
+      borderColor={borderColor}
+      textColor={textColor}
+      // common token props
+      as={as}
+      hideRemoveButton={hideRemoveButton || !handleRemove}
+      handleRemove={handleRemove}
+      id={id?.toString()}
+      isSelected={isSelected}
+      ref={forwardedRef}
+      text={text}
+      variant={variant}
+      {...rest}
+    >
+      <TokenTextContainer>{text}</TokenTextContainer>
+      {!hideRemoveButton && handleRemove ? (
+        <RemoveTokenButton
```

Comment:
Is nesting a button inside another button or link an accessibility issue?

Reply:
You addressed this in your next comment: we specifically avoid rendering a button inside of a button or link.